internal/dag/dag.go

```diff
@@ -191,6 +191,14 @@ type UpstreamValidation struct {
 	SubjectName string
 }
 
+// CACertificateKey stores the key name for the TLS validation secret cert
+const CACertificateKey = "ca.crt"
+
+// ValidationContextCACert returns the CA certificate from ValidationContext
+func (vc *ValidationContext) ValidationContextCACert() []byte {
+	return vc.CACertificate.Object.Data[CACertificateKey]
```

Comment:
If you check for `vc == nil`, then this function can be used to replace `upstreamValidationCACert`.

Reply:
Done.